### ISSUE TITLE
fetchOHLCV requires a since argument to be supplied along with the limit argument

### ISSUE THREAD

**Issue by mw130** [ccxt/ccxt#4136]
https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue

- OS: OS X Mojave
- Programming Language version: python 3
- CCXT version: 1.17.480
- Exchange: All
- Method: fetchOHLCV
I was using fetchOHLCV to get some exchange historical data, and now even when passing it with a 'since' argument, I get the error `    raise ExchangeError(self.id + ' fetchOHLCV requires a since argument to be supplied along with the limit argument')` for every single exchange. 
The call that triggers this is: 
`            ohlcv = client.fetch_ohlcv(coin_pair, timeframe='1m', since=then, limit=None)`
This was working fine before - I am not sure what has changed to make this throw an error, since the `since` is always being passed.

**Comment by kroitor** [comment]
@mw130 can you make a short, but a complete snippet (including the instantiation code and the call) to reproduce the issue?

```
> mbp:ccxt igorkroitor$ python examples/py/cli.py binance fetch_ohlcv ETH/BTC 1m 1541462400000
[[1541462400000, 0.032561, 0.032571, 0.03256, 0.032568, 109.788],
 [1541462460000, 0.032563, 0.032597, 0.032563, 0.03259, 117.847],
 [1541462520000, 0.03259, 0.032591, 0.032568, 0.032588, 42.569],
 [1541462580000, 0.032583, 0.032589, 0.03257, 0.032571, 49.743],
 [1541462640000, 0.032578, 0.032586, 0.032567, 0.032568, 59.68],
 [1541462700000, 0.032579, 0.032579, 0.032558, 0.032558, 124.049],
 [1541462760000, 0.03257, 0.032601, 0.032558, 0.03257, 46.496],
...
```

**Comment by mw130** [comment]
My apologies @kroitor , this is for bitz exchange only, not the others. As a side  note, to note open a new issue, do you know if the fact that you can't fetch historical data on Kraken a ccxt issue or an issue on their side? If you call 1m data from 2016, instead of returning 1m data from then until the limit of datapoints, it will return the earliest it can until present (which is about a day)

**Comment by kroitor** [comment]
> My apologies @kroitor , this is for bitz exchange only, not the others.

Oh... The warning isn't quite clear, you should read it as: *"For bitz, fetchOHLCV requires a limit argument if the since argument is specified."* I've changed the phrasing, so it should be clear from now.

> As a side note, to note open a new issue, do you know if the fact that you can't fetch historical data on Kraken a ccxt issue or an issue on their side? 

Yes, and, as long as it's documented, it isn't an issue on either side ) You should read the Manual really carefully: https://github.com/ccxt/ccxt/wiki/Manual#ohlcv-candlestick-charts as it's explained there:

> *There's a limit on how far back in time your requests can go. Most of exchanges will not allow to query detailed candlestick history (like those for 1-minute and 5-minute timeframes) too far in the past. They usually keep a reasonable amount of most recent candles, like 1000 last candles for any timeframe is more than enough for most of needs. You can work around that limitation by continuously fetching (aka REST polling) latest OHLCVs and storing them in a CSV file or in a database.*

In other words, if the exchange has a limit of 10 on any candle size, then for 1m-timeframe you will be able to fetch 10 last minutes only, for 1h - ten last hours and so on... So, what you described with Kraken, is pretty much expected. 

Hope this answers your questions.

**Comment by mw130** [comment]
@kroitor 

> Yes, and, as long as it's documented, it isn't an issue on either side ) You should read the Manual really carefully: https://github.com/ccxt/ccxt/wiki/Manual#ohlcv-candlestick-charts as it's explained there:
> 
> There's a limit on how far back in time your requests can go. Most of exchanges will not allow to query detailed candlestick history (like those for 1-minute and 5-minute timeframes) too far in the past. They usually keep a reasonable amount of most recent candles, like 1000 last candles for any timeframe is more than enough for most of needs. You can work around that limitation by continuously fetching (aka REST polling) latest OHLCVs and storing them in a CSV file or in a database.
> 
> In other words, if the exchange has a limit of 10 on any candle size, then for 1m-timeframe you will be able to fetch 10 last minutes only, for 1h - ten last hours and so on... So, what you described with Kraken, is pretty much expected.
> 
> 

Does this mean that you can't scrape 1m historical data on exchanges?

**Comment by mw130** [comment]
@kroitor sorry, just to follow up, if this is the case, is there a list of which exchanges will allow querying 1/5m historical data and which will just return empty lists?

**Comment by kroitor** [comment]
> Does this mean that you can't scrape 1m historical data on exchanges? That only with continuous fetching you could start building up data from today to the future but not the past?

It means, you can only scrape the entire history from very few exchanges, and most of them will serve a limited version of it anyways. So, yes, if you're looking for a generalized solution, I'm afraid you will have to start gathering that data or obtain it from whatever paid or free resources and services you can find online... That is a limitation coming from exchanges' side, obviously, we don't have a reason to limit the history anywhere in the library and we would be happy if they could serve more.

**Comment by mw130** [comment]
I see. The last thing I would ask then is, do you know which exchanges do allow most of the history? Do you know how places like Cryptotick.com can get the data? Your help is much appreciated.
